service/matching/physical_task_queue_manager.go

```diff
@@ -545,8 +547,13 @@ func (c *physicalTaskQueueManagerImpl) ensureRegisteredInDeployment(
 	defer c.deploymentLock.Unlock()
 
 	if c.deploymentRegistered {
-		// already done
-		return nil
+		// deployment already registered
+		return deployment.ErrTaskQueueExistsInDeployment
```

Comment:
why the error here?

Reply:
eh, actually you are right we can have multiple successive polls we don't wanna show this to the customer - I went down the rabbit hole of trying to be tooo clear to our customers :(